src/library/trackcollection.cpp

```diff
@@ -547,7 +547,7 @@ TrackPointer TrackCollection::getTrackById(
 
 TrackPointer TrackCollection::getTrackByRef(
         const TrackRef& trackRef) const {
-    DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
+    //DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
```

Comment:
@uklotzde this is necessary to retrieve the track for the AsyncImageResponse thread. Should I revert?

Reply:
Oh, sorry, forgot that we are in another thread context here :see_no_evil: 

We had introduced TrackLoader for this purpose. Would this help?

Reply:
Maybe not, because the result is dispatched by a signal and requires an event loop.